File: 프로그래머스/unrated/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.java
import java.util.ArrayList;
import java.util.List;
class Solution {
    public int solution(int a, int d, boolean[] included) {
        List<Integer> common_difference = new ArrayList<>();
		List<Integer> trueIdx = new ArrayList<>();
		int temp, result = 0;

		// 등차수열 만들고, true 의 인덱스를 구한다.
		for (int i = 0; i < included.length; i++) {
			common_difference.add(a + (i * d));
			if (included[i])
				trueIdx.add(i);
		}

		// 등차수열에서 true 의 인덱스를 뽑아 더한다.
		for (int index : trueIdx) {
			result += common_difference.get(index);
		}
		return result;
    }
}

File: 프로그래머스/unrated/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.js
function solution(a, d, included) {
    let comm = [];
    let trueIdx = [];
    let result = 0;
    
    for (let i = 0; i < included.length; i++) {
        comm.push(a + (i * d));
        if (included[i]) trueIdx.push(i);
    }

    for (let i = 0; i < trueIdx.length; i++) {
        result += comm[trueIdx[i]];
    }
    return result;
}